src/hooks/useServerApi.ts

```
@@ -0,0 +1,242 @@
+        additionalHeaders: updatedHeaders,
+      });
+    } catch (err) {
+      if (err instanceof APIError || err instanceof ClientError) {
+        throw err;
+      }
+      console.error('Unexpected error:', err);
+      throw new Error('An unexpected error occurred');
+    }
+  };
+
+  const getApiAuthenticated = async <T, P extends Record<string, string> = {}>(
```

Biome (1.9.4)

[error] Don't use '{}' as a type.

Prefer explicitly define the object shape. '{}' means "any non-nullable value".

(lint/complexity/noBannedTypes)